e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/add-to-an-existing-payment.spec.js

```diff
@@ -149,4 +164,111 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can add fee records to existing paymen
       pages.utilisationReportAddToAnExistingPaymentPage.errorSummary().contains('Select a payment to add the fee or fees to');
     });
   });
+
+  it('should add the selected fees to the payments and update the fee record status after clicking the continue button', () => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord]);
+
+    navigateToAddToExistingPaymentScreenForFirstFeeRecord();
+
+    pages.utilisationReportAddToAnExistingPaymentPage.continueButton().click();
+
+    pages.utilisationReportPage.premiumPaymentsTab.getPaymentLink(PAYMENT_ID_ONE).should('contain', 'GBP 450.00');
+    pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable
+      .status(FEE_RECORD_ID_ONE)
+      .invoke('text')
+      .then((text) => {
+        expect(text.trim()).to.equal('MATCH');
+      });
+  });
+
+  describe('when user navigates away', () => {
+    const FEE_RECORD_ID_THREE = '33';
+    const FEE_RECORD_ID_FOUR = '44';
+
+    beforeEach(() => {
+      const thirdFeeRecord = FeeRecordEntityMockBuilder.forReport(report)
+        .withId(FEE_RECORD_ID_THREE)
+        .withFacilityId('33333333')
+        .withExporter('Exporter 3')
+        .withPaymentCurrency(PAYMENT_CURRENCY)
+        .withFeesPaidToUkefForThePeriod(300)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .withPaymentExchangeRate(0.5)
+        .withStatus(FEE_RECORD_STATUS.TO_DO)
+        .build();
+      const fourthFeeRecord = FeeRecordEntityMockBuilder.forReport(report)
+        .withId(FEE_RECORD_ID_FOUR)
+        .withFacilityId('44444444')
+        .withExporter('Exporter 4')
+        .withPaymentCurrency(PAYMENT_CURRENCY)
+        .withFeesPaidToUkefForThePeriod(400)
+        .withFeesPaidToUkefForThePeriodCurrency('EUR')
+        .withPaymentExchangeRate(0.5)
+        .withPayments([firstPayment])
+        .withStatus(FEE_RECORD_STATUS.DOES_NOT_MATCH)
+        .build();
+
+      cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+      cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+      cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+      cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord, thirdFeeRecord, fourthFeeRecord]);
+    });
+
+    describe('by clicking the back button', () => {
+      it('should redirect the user to the premium payments page and persist the selected fees', () => {
```

Comment:
Could we split up the redirect and persistence assertions into their own "it's"? Could have a `before`/`beforeEach` for prerequisites